sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RetryUtils.java

```diff
@@ -127,7 +132,7 @@ private static void startBackgroundAddressRefresh(
         AddressSelector addressSelector) {
 
         addressSelector.resolveAddressesAsync(request, true)
-                       .publishOn(Schedulers.elastic())
+                       .publishOn(Schedulers.boundedElastic())
```

Comment:
Reactor deprecated `elastic()` thread scheduler in their current release, its good to update it here. 
But there is a difference between `elastic()` and `boundedElastic()`. 
`Elastic` has unbounded number of threads, so basically creating a new thread every time a new task is given to it, while `BoundedElastic` has bounded number of threads - `10 * Runtime.getRuntime().availableProcessors()`
It would be good to verify that this boundedness doesn't create any contentions for our SDK and its use case.

Reply:
Sure - we will need to go through CTL run anyway for this change, correct?

Reply:
Yes, I believe so.